textrepo-app/src/main/java/nl/knaw/huc/service/version/JdbiVersionService.java

```diff
@@ -92,27 +92,32 @@ public Version get(UUID id) {
 
   @Override
   public void delete(UUID id) {
-    var version = versions().find(id);
-    versions().delete(id);
-    version.ifPresent(v -> tryDeletingContents(id, v.getContentsSha()));
-  }
-
-  /**
-   * Deletes contents when its sha is not linked to any version
-   */
-  private void tryDeletingContents(@Nonnull UUID id, String sha) {
-    try {
-      contents().delete(sha);
-      log.info("Deleted contents of version {} by sha {}", id, sha);
-    } catch (JdbiException ex) {
-      if (violatesConstraint(ex, VERSIONS_CONTENTS_SHA)) {
-        log.info("Not deleting contents of version {} because sha {} is still in use", id, sha);
-      } else {
-        throw ex;
+    var deletedIsLatestVersion = new AtomicBoolean();
```

Comment:
omzetten in resultobject en juiste jdbi.inTransaction die dat teruglevert inzetten?

Reply:
Zie 9f507b1

Reply:
Beter zo! 👍